pkg/provisioningv2/rke2/planner/initnode.go

```diff
@@ -65,35 +65,35 @@ func (p *Planner) findAndDesignateFixedInitNode(rkeControlPlane *rkev1.RKEContro
 			err := p.clearInitNodeMark(entry)
 			if err != nil && !errors.Is(err, generic.ErrSkip) {
 				// if we received a strange error attempting to clear the init node mark
-				return false, "", err
+				return false, "", nil, err
 			} else if errors.Is(err, generic.ErrSkip) {
 				cachesInvalidated = true
 			}
 		}
 		if cachesInvalidated {
-			return false, "", generic.ErrSkip
+			return false, "", nil, generic.ErrSkip
 		}
 		if err := p.setInitNodeMark(entries[0]); err != nil && !errors.Is(err, generic.ErrSkip) {
-			return false, "", err
+			return false, "", nil, err
 		}
 		rkeControlPlane = rkeControlPlane.DeepCopy()
 		rkeControlPlane.Labels[rke2.InitNodeMachineIDDoneLabel] = "true"
 		_, err := p.rkeControlPlanes.Update(rkeControlPlane)
 		if err != nil {
-			return false, "", err
+			return false, "", nil, err
 		}
 		// if we set the designated init node on this iteration, return an errSkip so we know our cache is invalidated
-		return true, entries[0].Metadata.Annotations[rke2.JoinURLAnnotation], generic.ErrSkip
+		return true, entries[0].Metadata.Annotations[rke2.JoinURLAnnotation], entries[0], generic.ErrSkip
 	}
 	logrus.Debugf("rkecluster %s/%s: designated init node %s found", rkeControlPlane.Namespace, rkeControlPlane.Spec.ClusterName, fixedMachineID)
-	return true, entries[0].Metadata.Annotations[rke2.JoinURLAnnotation], nil
+	return true, entries[0].Metadata.Annotations[rke2.JoinURLAnnotation], entries[0], nil
 }
 
 // findInitNode searches the given cluster for the init node. It returns a bool which is whether an init node was
 // found, the init node join URL, and an error for a few conditions, i.e. if multiple init nodes were found or if there
```

Comment:
Can you update the godoc comment to reflect that the function is now returning the node entry as well for the found init node?